similaritymeasures/similaritymeasures.py

```diff
@@ -879,3 +879,125 @@ def dtw_path(d):
     path = np.array(path)
     # reverse the order of path, such that it starts with [0, 0]
     return path[::-1]
+
+def mae(exp_data, num_data, metric='euclidean', **kwargs):
+    """
+    Compute the Mean Absolute distance.
+
+    This computes the mean of absolute values of the distances 
+    between 2 corresponding points on the curve. This can use all 
+    distance metrics that are available in scipy.spatial.distance.cdist.
+
+    Parameters
+    ----------
+    exp_data : array_like
+        Curve from your experimental data. exp_data is of (M, N) shape, where
+        M is the number of data points, and N is the number of dimmensions
+    num_data : array_like
+        Curve from your numerical data. num_data is of (P, N) shape, where P
+        is the number of data points, and N is the number of dimmensions
+    metric : str or callable, optional
+        The distance metric to use. Default='euclidean'. Refer to the
+        documentation for scipy.spatial.distance.cdist. Some examples:
+        'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation',
+        'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'kulsinski',
+        'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto', 'russellrao',
+        'seuclidean', 'sokalmichener', 'sokalsneath', 'sqeuclidean',
+        'wminkowski', 'yule'.
+    **kwargs : dict, optional
+        Extra arguments to `metric`: refer to each metric documentation in
+        scipy.spatial.distance.
+
+        Some examples:
+
+        p : scalar
+            The p-norm to apply for Minkowski, weighted and unweighted.
+            Default: 2.
+
+        w : ndarray
+            The weight vector for metrics that support weights (e.g.,
+            Minkowski).
+
+        V : ndarray
+            The variance vector for standardized Euclidean.
+            Default: var(vstack([XA, XB]), axis=0, ddof=1)
+
+        VI : ndarray
+            The inverse of the covariance matrix for Mahalanobis.
+            Default: inv(cov(vstack([XA, XB].T))).T
+
+        out : ndarray
+            The output array
+            If not None, the distance matrix Y is stored in this array.
+
+    Retruns
+    -------
+    r : float
+        MAE.
+    """
+    c = distance.cdist(exp_data, num_data, metric=metric, **kwargs)
+    mae = 0
+    for i in range(0, len(c)):
+        mae += abs(c[i, i])
+    mae /= len(c)
+    return mae
+
```

Comment:
I think this PR should add 3 methods. The 3 methods are the following:
1. `pnorm_distance`
2. `mae`
3. `mse`

You'll create `pnorm_distance` to return the sum of `scipy.sptial.distance_matrix`, and you'll want the user to be able to optionally specify both `p` and `threshold` (sticking to scipy's defaults).

Then both `mae` and `mse` will call pnorm_distance for their special `p` cases (and also computing the mean).

Reply:
I dont think that the mean of distance_matrix(p=2) = MSE,
Here is a counter example,

Let

exp_data = [(0,0), (1,0)]
num_data = [(0,1), (1,2)]

(Note: Since only the y coordinate is different (for errors)
euclidian distance is equal to manhatten distance
and minkowski distance for all values of p)

Dist(x, y) refers to the distance between points x, y

Here,
MAE = (|Dist((0,0), (0,1))| + |Dist((1,0), (1,2))|)/2
    = (1 + 2)/2
    = 1.5
MSE = (|Dist((0,0), (0,1))|^2 + |Dist((1,0), (1,2))|^2)/2
    = ((1^2) + (2^2))/2
    = (1 + 4)/2
    = 2.5
distance_matrix(p=2) = [1.	, 1.41
				       2.23	, 2.  ]
Mean of Diagonal = 1.5
distance_matrix(p=1) = [1.	, 2.
				       3.	, 2.  ]
Mean of Diagonal = 1.5

So since Mean of distance_matrix(p=2) = 1.5 and MSE = 2.5
Mean of distance_matrix(p=2) is not equal to MSE

I hope I have understood you correctly. If not please let me know.

Reply:
You are absolute right. The `scipy.spatial.distance_matrix` does not do what I expected.

I think it's fine if you don't implement `pnorm_distance` as it doesn't really make sense here.

Can you still implement mae and mse without using scipy.spatial.cdist?

I think this could look like
```
# mae
np.abs(x-y).mean()
# mse
np.square(x-y).mean()
```